File: Python/range-sum-query-immutable.py
class NumArray:
    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        self.accu = [0]
        for num in nums:
            self.accu.append(self.accu[-1] + num)

    def sumRange(self, i, j):
        """
        :type i: int
        :type j: int
        :rtype: int
        """
        return self.accu[j + 1] - self.accu[i]

File: Swift/range-sum-query-immutable.swift
class NumArray {
    private var accu: [Int]

    init(_ nums: [Int]) {
        accu = [0]
        for num in nums {
            accu.append(accu.last! + num)
        }
    }

    func sumRange(_ i: Int, _ j: Int) -> Int {
        return accu[j + 1] - accu[i]
    }
}
